spec/features/movies/top_rated_results_spec.rb

```diff
@@ -0,0 +1,64 @@
+require "rails_helper"
+
+RSpec.describe "Top Rated Results Spec" do
+  before :each do
+    @user = User.create!(user_id: "100000000000000000000", name: "John Smith")
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
+  end
+
+  describe "As an authorized user on the Movies Results Index page" do
+    before :each do
+      visit movies_path
+    end
+
+    it "There is a button to discover top rated movies" do
+      within ".top-rated-movies" do
+        expect(page).to have_button "Find Top Rated Movies"
+      end
+    end
+
+    it "a text field with a 'Find Movies' button" do
+      within ".find-movies" do
+        expect(page.has_field? "movie-title-keywords").to be_truthy
+        expect(page).to have_button "Find Movies"
+      end
+    end
+  end
+
+  describe "When I click the button to discover top rated movies" do
+    it "There are 40 results of Movies" do
+      visit movies_path
+      click_button "Find Top Rated Movies"
+      expect(current_path).to eq("/movies")
+
+      within ".results" do
+        expect(page).to have_selector("article[class*=movie]", count: 40)
+      end
+    end
+  end
+
+  # describe "Each movie result has" do
```

Comment:
We need to learn more about mocks and stubs to finish this spec.
The API is successfully being consumed and the view renders appropriately.
However, since the movie resources from the API are constantly changing, our test will break unless we use mocks and stubs.

Reply:
Reference the [screenshot](https://github.com/Kathybui732/viewing_party/pull/37#issuecomment-678032586)